### PR TITLE
py-pip-tools: bump to latest 6.12.3

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: bluebrain/repo-bluebrain/packages/py-pip-tools/package.py [hunk 1]
```diff
@@ -3,24 +3,20 @@
...
-    version("6.3.0", sha256="1835a1848bdfb22b2b6e5d10d630844ff5ee15e24b6c3bf92319c76f205d347f")
-
-    depends_on("py-click@7:")
-    depends_on("py-setuptools@45:", type="build")
-    depends_on("py-setuptools-scm+toml@6.2:", type="build")
-    depends_on("py-pip@20.3:")
-    depends_on("py-build")
-    depends_on("py-wheel")
+    depends_on("py-click@8:", type=("build", "run"))
+    depends_on("py-setuptools@63:", type="build")
```
Comment (heerener) [inline bluebrain/repo-bluebrain/packages/py-pip-tools/package.py]:
```suggestion
    depends_on("py-setuptools@63:", type=("build", "run"))
```

File: bluebrain/repo-bluebrain/packages/py-pip-tools/package.py [hunk 2]
```diff
@@ -3,24 +3,20 @@
...
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-from spack import *
+from spack.package import *
 
 
 class PyPipTools(PythonPackage):
     """pip-tools keeps your pinned dependencies fresh."""
 
     homepage = "https://pip-tools.rtfd.io"
```
Comment (heerener) [inline bluebrain/repo-bluebrain/packages/py-pip-tools/package.py]:
```suggestion
    homepage = "https://github.com/jazzband/pip-tools/"
```

**Review by heerener — CHANGES_REQUESTED** [review]

**Review by heerener — DISMISSED** [review]

**Comment by tristan0x** [comment]
> To test your PR, use the following on BlueBrain5:
> 
> ```shell
> unset MODULEPATH
> . /gpfs/bbp.cscs.ch/ssd/apps/bsd/pulls/1874/config/modules.sh
> module load unstable
> ```
> 
> Please test the following updated modules:
> 
> * `steps/4.0.0`
> 
> The following module tests failed:
> 
> * [spykfunc](https://bbpgitlab.epfl.ch/hpc/module-testing/-/jobs/590674)

Can we bypass the failing spykfunc module and merge this PR @heerener ? This PR is kind of urgent and blocks many projects

**Comment by matz-e** [comment]
Needs another rebase, sorry

**Comment by tristan0x** [comment]
> Needs another rebase, sorry

No pb. This is life, not paradise.
Thanks you for your effort and support.

File: bluebrain/repo-bluebrain/packages/py-pip-tools/package.py (hunk 1 above)
Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/py-pip-tools/package.py]: Why was this resolved?

This is in `pyproject.toml`:
```
dependencies = [
  # direct dependencies
  "build",
  "click >= 8",
  "pip >= 22.2",
  # indirect dependencies
  "setuptools", # typically needed when pip-tools invokes setup.py
  "wheel", # pip plugin needed by pip-tools

]
```
Seems a clear run dependency to me...

https://github.com/jazzband/pip-tools/blob/6.12.3/pyproject.toml#L37-L47

**Review by matz-e — DISMISSED** [review]

File: bluebrain/repo-bluebrain/packages/py-pip-tools/package.py (hunk 1 above)
Comment (heerener) [reply bluebrain/repo-bluebrain/packages/py-pip-tools/package.py]: I have an impression that both the changes I suggested were undone, possibly by the force push?

**Review by heerener — APPROVED** [review]

**Pull request merged**